### DISCUSSION TITLE
support for version argument on generateschema command

### DISCUSSION THREAD

**Discussion by erancihan** [encode/django-rest-framework#8551] [Ideas & Suggestions] [answered]
Hi everyone,

As it is now, [generateschema](https://github.com/encode/django-rest-framework/blob/master/rest_framework/management/commands/generateschema.py) command does not support `--version` argument. I was looking around the code and I found that the [SchemaGenerator](https://github.com/encode/django-rest-framework/blob/master/rest_framework/schemas/openapi.py) class currently supports version input if I am correct.

I do not know whether this is intentional or not, however if it is not, I would be more than happy to try and add it.

EDIT:
It is in order to add _my_ Current and Expected output for the generated `openapi-schema.yml` file.

**Current:**
```yaml
openapi: 3.0.2
info:
  title: My API
  version: ''
  description: My API Documentation
paths:
...
```

**Expected:**
```yaml
openapi: 3.0.2
info:
  title: My API
  version: 1.0.0
  description: My API Documentation
paths:
...
```

I am running the following command to generate the yaml file:
```bash
python manage.py generateschema \
--title "My API" \
--description "My API Documentation" \
--format openapi \
--file path/to/static/openapi-schema.yml
```
including `--version` arg does not work, as it is not supported as of `3.13.1`

**Comment by Azanul** [comment]
What is the end result you are looking for here?
If you want version to be mentioned in the generated schema document then changes will be first needed in Document Interface of core-api. Please refer to [the issue in core-api repo](https://github.com/core-api/core-api/issues/49).

**Reply by erancihan** [reply]
The end result I am expecting is something like this for the generated yaml file:
```yaml
openapi: 3.0.2
info:
  title: My API
  version: 1.0.0
  description: My API Documentation
paths:
...
```

However, as things are know, there is no apparent way to fill out the `version` field under `info`.
What I am looking for is to add an argument that supplies and sets this field.

With current version, the generated doc looks sth like this:
```yaml
openapi: 3.0.2
info:
  title: My API
  version: ''
  description: My API Documentation
paths:
...
```

**Reply by erancihan** [reply]
I've checked the issue you have linked, however it seems that that repository is... archived.

I don't know if there is much we can do on that part...

**Reply by Azanul** [reply]
Your concern will be resolved in the next release.
> We're planning to iteratively work towards OpenAPI becoming the standard schema representation. This will mean that the coreapi dependency will gradually become removed, and we'll instead generate the schema directly, rather than building a CoreAPI Document object.
[Reference](https://www.django-rest-framework.org/community/3.9-announcement/#whats-next)

**Reply by erancihan** [reply]
Very well then, perhaps my problems will indeed be magically solved by the next release :D

This is not exactly a great problem to me, however I would have loved for that feature to exists.
If there is already a work on it, then all is well.

I'm looking forward to next release.
I'd like to keep the discussion open just in case.

**Reply by Azanul** [reply]
I'm also currently looking for a PR or an issue for attaining this and trying a couple things myself. I'll update here if I find any.

**Reply by Azanul** [reply]
I think the functionality that you're looking for is just two lines away. Do you want to do it or may I?

**Reply by erancihan** [reply]
I'll get on to it!

**Reply by Azanul** [reply]
By the way, this argument will serve any purpose as anyone who want to mention the version can just go and add it to the file manually. The amount of work done remains the same either way 😂. 
If you implement it then the use will just have to mention the version in command line argument instead of the generated document.
I think this is why it wasn't prioritised before.

**Reply by erancihan** [reply]
I get your point :D

I feel the need for it as I want to be able to update version on the fly in GitHub Actions, and have it reflect on the docs when deployed.
I found it less work to maintain version number through actions instead of manually going and updating it, when there are multiple PRs on and about 😅

**Reply by Azanul** [reply]
Now it is making sense to me. Thanks a lot for clarifying.
Don't forget mark this as answered once your PR for the same is merged.

**Reply by Azanul** [reply]
@erancihan have you not raised the PR yet?

**Reply by erancihan** [reply]
oh yes, that's done

**Comment by erancihan** [comment, marked as answer]
#8663
